File: packages/durable-streams/packages/client-java/src/main/java/com/durablestreams/exception/StreamNotFoundException.java
package com.durablestreams.exception;

/**
 * Thrown when a stream does not exist (404).
 */
public class StreamNotFoundException extends DurableStreamException {
    private final String url;

    public StreamNotFoundException(String url) {
        super("Stream not found: " + url, 404, "NOT_FOUND");
        this.url = url;
    }

    /**
     * Returns the URL of the stream that was not found.
     */
    public String getUrl() {
        return url;
    }
}

File: packages/durable-streams/packages/client-php/src/Exception/StreamNotFoundException.php
<?php

declare(strict_types=1);

namespace DurableStreams\Exception;

class StreamNotFoundException extends DurableStreamException
{
    public function __construct(string $url, ?\Throwable $previous = null)
    {
        parent::__construct(
            "Stream not found: {$url}",
            'NOT_FOUND',
            404,
            [],
            $previous
        );
    }
}
